filemage/tests/test_filemage.py

```diff
@@ -0,0 +1,491 @@
+
+
+def mockPrcoessIterServicesUp(attrs=None, ad_value=None):
+    def procIterFormat(proc):
+        proc.info = proc.as_dict(attrs=attrs, ad_value=ad_value)
+        return proc
+
+    yield procIterFormat(MockProcess(name='systemd', cmdline=['/sbin/init']))
+    yield procIterFormat(
+        MockProcess(
+            name='postgres',
+            cmdline=[
+                '/usr/lib/postgresql/12/bin/postgres',
+                '-D',
+                '/var/lib/postgresql/12/main',
+                '-c',
+                'config_file=/etc/postgresql/12/main/postgresql.conf',
+            ],
+        )
+    )
+    yield procIterFormat(
+        MockProcess(
+            name='gateway',
+            cmdline=[
+                '/opt/filemage/bin/gateway',
+                '-c',
+                '/etc/filemage/config.yml',
+            ],
+        )
+    )
+
+
+def mockRequestsGetMetricsDown(*args, **kwargs):
+    return MockResponse({'message': 'Session expired.'}, 401)
+
+
+def mockRequestsGetMetricsUp(*args, **kwargs):
+    return MockResponse(
+        [
+            {'timestamp': '2022-12-09T17:00:10Z', 'path': '/tmp/example.csv', 'user': 'example', 'operation': 'put'},
+            {'timestamp': '2022-12-09T17:00:10Z', 'path': '/tmp/example.csv', 'user': 'example', 'operation': 'rmdir'},
+        ],
+        200,
+    )
+
+
+@pytest.mark.unit
+def test_bad_instance(dd_run_check, aggregator, bad_instance):
+    with pytest.raises(ConfigurationError):
+        c = FilemageCheck('filemage', {}, [bad_instance])
+        dd_run_check(c)
+
+
+@pytest.mark.unit
+def test_good_instance(dd_run_check, aggregator, good_instance):
+    c = FilemageCheck('filemage', {}, [good_instance])
+    dd_run_check(c)
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+@mock.patch('datadog_checks.filemage.check.requests.get', mockRequestsGetMetricsUp)
+@mock.patch('datadog_checks.filemage.check.psutil.process_iter', mockPrcoessIterServicesUp)
+def test_check_coverage(dd_run_check, aggregator, good_instance):
+    c = FilemageCheck('filemage', {}, [good_instance])
+    dd_run_check(c)
+    for check in EXPECTED_CHECKS:
+        aggregator.assert_service_check(check)
+    aggregator.assert_no_duplicate_service_checks()
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+@mock.patch('datadog_checks.filemage.check.requests.get', mockRequestsGetMetricsUp)
+@mock.patch('datadog_checks.filemage.check.psutil.process_iter', mockPrcoessIterServicesUp)
+def test_metric_coverage(dd_run_check, aggregator, good_instance):
+    c = FilemageCheck('filemage', {}, [good_instance])
+    dd_run_check(c)
+    for metric in FilemageCheck.FTP_TRACKED_METRICS:
+        aggregator.assert_metric(metric)
+    aggregator.assert_no_duplicate_metrics()
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(FilemageCheck.TRACKED_METRICS_META)
```

Comment:
You should change this back to `get_metadata_metrics()` too as per the other comments.

Reply:
resolved in f136f6a01f0efedadc412dd0badb3f3d0b429fbc